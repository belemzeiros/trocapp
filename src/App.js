import React from 'react';

class App extends React.Component {
  render() {
    return <div>Ol√° Mundo!</div>;
  }
}

export default App;
