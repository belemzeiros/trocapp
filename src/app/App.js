import React from 'react';
import { hot } from 'react-hot-loader';
import { Helmet } from 'react-helmet';

import renderRoutes from '@/routes/render';
import routes from '@/routes';

import CssBaseline from '@material-ui/core/CssBaseline';

class App extends React.Component {
  componentDidMount() {
    // Remove styles gerado pelo SSR
    const jssStyles = document.getElementById('jss-server-side');
    if (jssStyles && jssStyles.parentNode) {
      jssStyles.parentNode.removeChild(jssStyles);
    }
  }

  render() {
    return (
      <React.Fragment>
        <CssBaseline />
        <Helmet defaultTitle="Trocas e Doações" titleTemplate="%s | Trocapp" />
        {renderRoutes(routes)}
      </React.Fragment>
    );
  }
}

export default hot(module)(App);
