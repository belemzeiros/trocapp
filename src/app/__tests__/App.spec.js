import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import App from '../App';

describe('App', () => {
  const shallow = createShallow();

  it('App will mount correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should remove jss-server-side from DOM', () => {
    const jssStyleNode = document.createElement('style');
    jssStyleNode.id = 'jss-server-side';
    document.body.appendChild(jssStyleNode);
    expect(document.getElementById('jss-server-side').tagName).toEqual('STYLE');
    shallow(<App />);
    expect(document.getElementById('jss-server-side')).toBeFalsy();
  });
});
