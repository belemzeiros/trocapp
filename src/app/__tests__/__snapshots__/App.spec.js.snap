// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App will mount correctly 1`] = `
<React.Fragment>
  <WithStyles(CssBaseline) />
  <HelmetWrapper
    defaultTitle="Trocas e Doações"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="%s | Trocapp"
  />
  <Switch>
    <Route
      exact={true}
      key="0"
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      key="1"
      path="/sobre"
      render={[Function]}
    />
    <Route
      exact={true}
      key="2"
      path="/detalhe-produto"
      render={[Function]}
    />
    <Route
      key="3"
      render={[Function]}
    />
  </Switch>
</React.Fragment>
`;
